microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -0,0 +1,242 @@
+        """,
+        requirement=simple_requirement(
+            supported_features=[Sriov],
+        ),
+        priority=1,
+    )
+    def check_dpdk_build(self, node: Node, log: Logger) -> None:
+        sriov_feature = node.features[Sriov]
+        sriov_is_enabled = sriov_feature.enabled()
+        log.info(f"Verify SRIOV is enabled: {sriov_is_enabled}")
+        assert_that(sriov_is_enabled).described_as(
+            "SRIOV was not enabled for this test node."
+        ).is_true()
+
+        # dump some info about the pci devices before we start
+        lspci = node.tools[Lspci]
+        log.info(f"LSPCI Info:\n{lspci.run().stdout}\n")
+
+        # enable hugepages (required by dpdk)
+        self._hugepages_init(node)
+        self._hugepages_enable(node, log)
+
+        # initialize testpmd tool (installs dpdk)
+        testpmd = DpdkTestpmd(node)
+        testpmd.install()
+
+        # initialize node nic info class (gathers info about nic devices)
+        node_nic_info = NodeNicInfo(node)
+        assert_that(len(node_nic_info)).described_as(
+            "Test needs at least 2 NICs on the test node."
+        ).is_greater_than_or_equal_to(2)
+        log.info(node_nic_info)
+
+        # grab a nic and run testpmd
+        test_nic = node_nic_info.get_nic(node_nic_info.get_upper_nics()[-1])
+        vdev_type = "net_vdev_netvsc"
+        testpmd_include_str = test_nic.testpmd_include(vdev_type)
+        testpmd_output = testpmd.run_with_timeout(testpmd_include_str, 30)
+        tx_pps = testpmd.get_tx_pps_from_testpmd_output(testpmd_output)
+        log.info(
+            f"TX-PPS:{tx_pps} from {test_nic._upper}/{test_nic._lower}:"
+            + f"{test_nic._pci_slot}"
+        )
+        assert_that(tx_pps).described_as(
+            f"TX-PPS ({tx_pps}) should have been greater than 2^20 (~1m) PPS."
+        ).is_greater_than(2 ** 20)
+
+    def _execute_expect_zero(self, node: Node, cmd: str) -> str:
+        result = node.execute(cmd, sudo=True, shell=True)
+        result.assert_exit_code()
+        return result.stdout
+
+    def _hugepages_init(self, node: Node) -> None:
+        mount = node.tools[Mount]
+        mount.mount("nodev", "/mnt/huge", type="hugetlbfs")
+        mount.mount("nodev", "/mnt/huge-1G", type="hugetlbfs", options="'pagesize=1G'")
+
+    def _hugepages_enable(self, node: Node, log: Logger) -> None:
+        self._execute_expect_zero(
+            node,
+            (
+                "sh -c 'echo 4096 > "
+                "/sys/devices/system/node/node0/hugepages/"
+                "hugepages-2048kB/nr_hugepages'"
+            ),
+        )
+        self._execute_expect_zero(
+            node,
+            (
+                "sh -c 'echo 1 > "
+                "/sys/devices/system/node/node0/hugepages/"
+                "hugepages-1048576kB/nr_hugepages'"
+            ),
+        )
+        node.execute("grep -i huge /proc/meminfo && ls /mnt/", shell=True)
```

Comment:
Does it need to verify the output?

Reply:
No, I'll just remove it I was previously printing the output just for validation but it's fine